src/main/cli/cli.c

```diff
@@ -1235,134 +1235,203 @@ static void cbCtrlLine(void *context, uint16_t ctrl)
+                port1PinioDtr = 0;
             } else {
-                pinioDtr = atoi(tok);
+                port1PinioDtr = atoi(tok);
 #endif /* USE_PINIO */
             }
-
+            break;
+        case 4:
+            port2Id = atoi(tok);
+            break;
+        case 5:
+            port2Baud = atoi(tok);
+            break;
+        case 6:
+            port2Mode = cliParseSerialMode(tok);
             break;
         }
         index++;
         tok = strtok_r(NULL, " ", &saveptr);
     }
 
-    if (baud == 0) {
+    if (port1Baud == 0 && port2Id == -1) {
         enableBaudCb = true;
     }
 
-    cliPrintf("Port %d ", id);
-    serialPort_t *passThroughPort;
-    serialPortUsage_t *passThroughPortUsage = findSerialPortUsageByIdentifier(id);
-    if (!passThroughPortUsage || passThroughPortUsage->serialPort == NULL) {
-        if (enableBaudCb) {
-            // Set default baud
-            baud = 57600;
-        }
-
-        if (!mode) {
-            mode = MODE_RXTX;
-        }
+    if (port1PinioDtr < 0 || port1PinioDtr > PINIO_COUNT) {
+        cliPrintLinef("Invalid PinIO number %d", port1PinioDtr);
+        return ;
+    }
 
-        passThroughPort = openSerialPort(id, FUNCTION_NONE, NULL, NULL,
-                                         baud, mode,
-                                         SERIAL_NOT_INVERTED);
-        if (!passThroughPort) {
-            cliPrintLine("could not be opened.");
-            return;
-        }
+    // Port checks
+    if (findSerialPortIndexByIdentifier(port1Id) == -1) {
+        cliPrintLinef("Invalid port1 %d", port1Id);
+    } else {
+        cliPrintLinef("Port1: %d ", port1Id);
+    }
 
-        if (enableBaudCb) {
-            cliPrintf("opened, default baud = %d.\r\n", baud);
+    if (port2Id != -1) {
+        if (findSerialPortIndexByIdentifier(port2Id) == -1) {
+            cliPrintLinef("Invalid port2 %d", port2Id);
+        } if (port2Id == port1Id) {
+            cliPrintLinef("Port1 and port2 are same");
+            return ;
         } else {
-            cliPrintf("opened, baud = %d.\r\n", baud);
+            cliPrintLinef("Port2: %d", port2Id);
         }
     } else {
-        passThroughPort = passThroughPortUsage->serialPort;
-        // If the user supplied a mode, override the port's mode, otherwise
-        // leave the mode unchanged. serialPassthrough() handles one-way ports.
-        // Set the baud rate if specified
-        if (baud) {
-            cliPrintf("already open, setting baud = %d.\n\r", baud);
-            serialSetBaudRate(passThroughPort, baud);
-        } else {
-            cliPrintf("already open, baud = %d.\n\r", passThroughPort->baudRate);
-        }
+        cliPrintLine("Port2: VCP port");
+    }
+
+    serialPort_t *port1 = NULL;
+    serialPort_t *port2 = cliPort;
+    serialPort_t **passthroughPorts[2] = { &port1,  &port2 };
+    int portIds[2] = { port1Id, 0  };
+    uint32_t bauds[2] = { port1Baud, 0 };
```

Comment:
The approach to use arrays for the initialisation is good, but it's better to define a struct to hold all the parameters you need, and then declare an array of this struct.
Also, I still think this copying of parameters is just making this harder to read and maintain. Rather than this, declare the array before parsing the input, and put the parameters directly into the array.